core/README.md

```diff
@@ -128,3 +180,108 @@ the [overview](../overview.md#example-landing-page) document.
+
+How you structure your graph of Catalogs can allow you to both group Collections together and create sub-groups
+of items within a Collection. 
+For example, your collections may be grouped so each represent a data product. This might mean
+you have a collection for each of Landsat 8 Collection 1, Landsat 8 Surface Reflectance, Sentinel-2 L1C, Sentinel-2
+L2A, Sentinel-5P UV Aerosol Index, Sentinel-5P Cloud, MODIS MCD43A4, MODIS MOD11A1, and MODIS MYD11A1. You can also
+present each of these as a catalog, and create parent catalogs for them that allow you to group together all Landsat, Sentinel, and MODIS catalogs.
+
+- / root catalog
+  - child -> /catalogs/landsat
+    - child -> /catalogs/landsat_7
+    - child -> /catalogs/landsat_8
+      - child -> /catalogs/landsat_8_c1
+      - child -> /catalogs/landsat_8_sr
+  - child -> /catalogs/sentinel
+    - child -> /catalogs/sentinel_2
+      - child -> /catalogs/sentinel_2_l1c
+      - child -> /catalogs/sentinel_2_l2a
+    - child -> /catalogs/sentinel_5p
+      - child -> /catalogs/sentinel_5p_uvai
+      - child -> /catalogs/sentinel_5p_cloud
+  - child -> /catalogs/modis
+    - child -> /catalogs/modis_mcd43a4
+    - child -> /catalogs/modis_mod11a1 
+    - child -> /catalogs/modis_myd11a1
+
+Each of these catalog endpoints could in turn be its own STAC API root, allowing an interface where users can
+search over arbitrary groups of collections without needing to explicitly know and name every collection in the
+search `collection`  query parameter. These catalogs-of-catalogs can be separated multiple ways, e.g. be
+per provider (e.g., Sentinel-2), per domain (e.g., cloud data), or per form of data (electro-optical, LIDAR, SAR).
+
+Going the other direction, collections can be sub-grouped into smaller catalogs. For example, this example
+groups a catalog of Landsat 8 Collection 1 items by path, row, and date (the path/row system is used by this
+product for gridding). 
+
+- / (root)
+  - /catalogs/landsat_8_c1
+    - /catalogs/landsat_8_c1/139
+      - /catalogs/landsat_8_c1/139_045
+        - /catalogs/landsat_8_c1/139_045_20170304
+          - /collection/landsat_8_c1/items/LC08_L1TP_139045_20170304_20170316_01_T1
+        - /catalogs/landsat_8_c1/139_045_20170305
+          - /collection/landsat_8_c1/items/LC08_L1TP_139045_20170305_20170317_01_T1
+      - /catalogs/landsat_8_c1/139_046
+        - /catalogs/landsat_8_c1/139_046_20170304
+          - /collection/landsat_8_c1/items/LC08_L1TP_139046_20170304_20170316_01_T1
+        - /catalogs/landsat_8_c1/139_046_20170305
+          - /collection/landsat_8_c1/items/LC08_L1TP_139046_20170305_20170317_01_T1
+
+If done in a consistent manner, these can also provide "templated" URIs, such that a user could directly request a
+specific path, row, and date simply by replacing the values in `/catalogs/landsat_8_c1/{path}_{row}_{date}`.
```

Comment:
Is there a way for the server to specify the templated URI for exposing children catalogs?  I'm thinking of a catalog-level field containing the appropriate URI template.  For example, the `/catalogs/landsat_8_c1` catalog would have a children URI template of `/catalogs/landsat_8_c1/{path}_{row}_{date}` while the `/catalogs/landsat_8_c1/139` catalog would have a children URI template of `/catalogs/landsat_8_c1/139_{row}_{date}`.  The base of the URI template is just the `self` link to the current catalog, while the latter half is the actual template.

This would be particularly powerful when combined with a collection-level summary defining all the possible values for these templates.

Reply:
+1 on this.  Also from an implementation side, where this is probably going to be generated dynamically, it would be useful for the catalog or collection to have that template specified within it so the server can use that to create the browse links dynamically.  As it stands now this would have to be specified outside of STAC...i.e., some implementation specific way

Reply:
OGC API Tiles has the convention of a `templated` link which could work here - https://github.com/opengeospatial/ogcapi-tiles/blob/336a40bb611b62dc9f3ee9954663bf1964bb903b/core/standard/requirements/tileset/REQ_tileset-description.adoc.

But I think this doesn't have to be a link.

Reply:
I'd just like to avoid that we put something into links that existing clients don't understand ;-)